File: chazm-api/src/main/java/runtimemodels/chazm/api/relation/Requires.java
package runtimemodels.chazm.api.relation;

import runtimemodels.chazm.api.Organization;
import runtimemodels.chazm.api.entity.Capability;
import runtimemodels.chazm.api.entity.Role;

/**
 * The {@linkplain Requires} interface defines the requires relation, where a {@linkplain Role} requires a {@linkplain Capability}, of an
 * {@linkplain Organization}.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
public interface Requires {
    /**
     * Returns the {@linkplain Role} of this {@linkplain Requires}.
     *
     * @return the {@linkplain Role} of this {@linkplain Requires}.
     */
    Role getRole();

    /**
     * Returns the {@linkplain Capability} of this {@linkplain Requires}.
     *
     * @return the {@linkplain Capability} of this {@linkplain Requires}.
     */
    Capability getCapability();
}

File: chazm-api/src/main/kotlin/runtimemodels/chazm/api/relation/Requires.kt
package runtimemodels.chazm.api.relation

import runtimemodels.chazm.api.Organization
import runtimemodels.chazm.api.entity.Capability
import runtimemodels.chazm.api.entity.Role

/**
 * The [Requires] interface defines a requires relation, where a [Role] requires a [Capability], of an
 * [Organization].
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
interface Requires {
    /**
     * The [Role] of this [Requires].
     */
    val role: Role

    /**
     * The [Capability] of this [Requires].
     */
    val capability: Capability
}
